driver/src/test/java/org/neo4j/driver/v1/tck/BoltTypeSteps.java

```diff
@@ -0,0 +1,445 @@
+import cucumber.runtime.CucumberException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.neo4j.driver.internal.InternalNode;
+import org.neo4j.driver.internal.InternalPath;
+import org.neo4j.driver.internal.InternalRelationship;
+import org.neo4j.driver.internal.value.NodeValue;
+import org.neo4j.driver.v1.Node;
+import org.neo4j.driver.v1.Path;
+import org.neo4j.driver.v1.Relationship;
+import org.neo4j.driver.v1.ResultCursor;
+import org.neo4j.driver.v1.Statement;
+import org.neo4j.driver.v1.Value;
+import org.neo4j.driver.v1.Values;
+
+import static java.lang.String.valueOf;
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.neo4j.driver.v1.Values.value;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.CypherStatementRunner;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.MappedParametersRunner;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.StatementRunner;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.StringRunner;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.asJavaArrayList;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.asJavaType;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.boltValueAsCypherString;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.boltValuetoJavaObject;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.databaseRunning;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.getBoltValue;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.getListFromString;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.getListOfTypes;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.getMapOfTypes;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.getPathOfEmptyNodesWithSize;
+import static org.neo4j.driver.v1.tck.TCKTestUtil.getRandomString;
+
+
+public class BoltTypeSteps
+{
+    private MappedParametersRunner mappedParametersRunner;
+    private StatementRunner statementRunner;
+    private StringRunner stringRunner;
+    private List<CypherStatementRunner> runners;
+    private Object expectedJavaValue;
+    private Value expectedBoltValue;
+    private List<Object> listOfObjects;
+    private Map<String,Object> mapOfObjects;
+
+    @Before
+    public void resetValues()
+    {
+        listOfObjects = new ArrayList<>();
+        mapOfObjects = new HashMap<>();
+        expectedJavaValue = null;
+        expectedBoltValue = null;
+        mappedParametersRunner = null;
+        statementRunner = null;
+        stringRunner = null;
+        runners = new ArrayList<>();
+    }
+
+    @After
+    public void print( Scenario sc )
+    {
+        if ( sc.isFailed() )
+        {
+            System.out.println( "Failed: " + sc.getName() );
```

Comment:
This is a great method, can we print more info for the failed tests? e.g. the expected result is a `5` of bolt `Integer`. (It would really helpful if we could get all the info, e.g. even `value` and `type` for the bug directly). Meanwhile, I would also suggest to skip the printout for success methods.